.github/workflows/e2e-matrix.yaml

```diff
@@ -1,13 +1,34 @@
 name: E2EMatrix
 on:
-  schedule:
-    - cron: '7 */8 * * *'
-  push:
-    branches: [main]
-  workflow_run:
-    workflows: [ApprovalComment]
-    types: [completed]
+  workflow_call:
+    inputs:
+      region:
+        type: string
+        default: "us-east-2"
+      k8s_version:
+        type: string
+        default: "1.27"
+    secrets:
+      SLACK_WEBHOOK_URL:
+        required: true
   workflow_dispatch:
+    inputs:
+      region:
+        type: choice
+        options:
+          - "us-east-2"
+          - "us-west-2"
+          - "us-west-1"
+        default: "us-east-2"
+      k8s_version:
+        type: choice
+        options:
+          - "1.23"
+          - "1.24"
+          - "1.25"
+          - "1.26"
+          - "1.27"
+        default: "1.27"
 # All jobs will run under the following conditions:
 #   1. Upstream Karpenter repo triggered the job by schedule, push, or dispatch
 #   2. Upstream Karpenter repo triggered the job by workflow_run, which requires maintainer check to succeed
```

Comment:
The resolve step needs to be moved up a level to the trigger level. I think we should pass in the `git_ref` into the `workflow_call` and resolve it in the `e2e-matrix-trigger`